peekingduck/nodes/output/screen.py

```diff
@@ -75,12 +75,34 @@ def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
         if self.first_run:
             cv2.moveWindow(self.window_name, self.window_loc["x"], self.window_loc["y"])
             self.first_run = False
-        if cv2.waitKey(1) & 0xFF == ord("q"):
+
+        return_key = self._wait_for_keyboard_input(inputs["filename"])
+
+        # Any key to close window
+        if return_key != -1:
             cv2.destroyWindow(self.window_name)
```

Comment:
Why is there a need to destroyWindow at each iteration?
(assuming showing a folder of images)
Why can't the same window be reused? (instead of freeing and allocating window resources each time)

Reply:
> 
`if return_key != -1:`
Should be applicable to close the window on multiple images without ending the pipeline. This is with the assumption that the user seldom press other keys in the video window.

To fix this assumption, check should be made to detect input type (video or image).